metadata_service/proxy/neo4j_proxy.py

```diff
@@ -236,7 +241,19 @@ def _exec_table_query(self, table_uri: str) -> Tuple:
             src = Source(source_type=table_records['src']['source_type'],
                          source=table_records['src']['source'])
 
-        return wmk_results, table_writer, timestamp_value, owner_record, tags, src, badges
+        prog_descriptions = self._extract_programmatic_descriptions_from_query(table_records)
+
+        return wmk_results, table_writer, timestamp_value, owner_record, tags, src, badges, prog_descriptions
+
+    def _extract_programmatic_descriptions_from_query(self, table_records: dict) -> list:
+        prog_descriptions = []
+        for prog_description in table_records.get('prog_descriptions', []):
+            LOGGER.info(prog_description)
+            source = prog_description['description_source']
```

Comment:
Just to confirm if `description_source` is guaranteed to be there.

Reply:
yes if it isn't there then something bad happened with data builder. But... better to be safe here. Thanks for pointing this out.